### PR TITLE
Prepare examples of prettier formatted code

### PR DESCRIPTION
**Pull request by JuliaGerasymenko**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
   ],
   "scripts": {
     "test": "npm run lint && metatests test/",
-    "lint": "eslint . --ignore-path .gitignore",
+    "lint": "eslint . --ignore-path .gitignore && prettier --debug-check \"**/*.js\" \"**/*.json\" \"**/*.md\" .eslintrc .prettierrc",
```
Comment (nechaido) [inline package.json]:
```suggestion
    "lint": "eslint . --ignore-path .gitignore && prettier --debug-check \"**/*.js\" \"**/*.json\" \"**/*.md\" .eslintrc .prettierrc  --ignore-path .gitignore",
```

File: lib/pg.provider.js [hunk 2]
```diff
@@ -395,34 +423,45 @@ class PostgresProvider extends StorageProvider {
...
-        errorCodes.INVALID_CREATION_OPERATION,
-        `Cannot create instances of category ${category} individually, it is ` +
-        `included in categories ${categorySchema.references.Include.join(', ')}`
-      ));
+      process.nextTick(
+        callback,
+        new GSError(
+          errorCodes.INVALID_CREATION_OPERATION,
+          `Cannot create instances of category ${category} individually, it is ` +
+            `included in categories ${categorySchema.references.Include.join(
```
Comment (belochub) [inline lib/pg.provider.js]:
```suggestion
            'included in categories' + categorySchema.references.Include.join(', ')
```

File: lib/pg.provider.js [hunk 3]
```diff
@@ -280,9 +293,11 @@ class PostgresProvider extends StorageProvider {
 
       this.pool.query(
         `SELECT ${escapedRightCategory}.* FROM ${escapedRightCategory} ` +
-        `INNER JOIN ${escapedManyTableName} ON ${escapedRightCategory}."Id" =` +
-        ` ${escapedManyTableName}.${escapeIdentifier(fieldName)}` +
-        ` WHERE ${escapedManyTableName}.${escapeIdentifier(leftCategory)} = $1`,
+          `INNER JOIN ${escapedManyTableName} ON ${escapedRightCategory}."Id" =` +
+          ` ${escapedManyTableName}.${escapeIdentifier(fieldName)}` +
+          ` WHERE ${escapedManyTableName}.${escapeIdentifier(
```
Comment (belochub) [inline lib/pg.provider.js]:
```suggestion
          ` WHERE ${escapedManyTableName}` +
          `.${escapeIdentifier(leftCategory)} = $1`
```

File: lib/pg.provider.js [hunk 4]
```diff
@@ -395,34 +422,44 @@ class PostgresProvider extends StorageProvider {
...
-        errorCodes.INVALID_CREATION_OPERATION,
-        `Cannot create instances of category ${category} individually, it is ` +
-        `included in categories ${categorySchema.references.Include.join(', ')}`
-      ));
+      process.nextTick(
+        callback,
+        new GSError(
+          errorCodes.INVALID_CREATION_OPERATION,
+          `Cannot create instances of category ${category} individually, it is ` +
+            `included in categories' + categorySchema.references.Include.join(', ')
```
Comment (belochub) [inline lib/pg.provider.js]:
```suggestion
            'included in categories' + categorySchema.references.Include.join(', ')
```

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+README.md
```
Comment (belochub) [inline .prettierignore]: Why is this ignored as well as the `README.ru.md` file below?

File: doc/api/jstp.md [hunk 6]
```diff
@@ -151,15 +151,15 @@ Select a record or multiple records in the category using provided jsql.
...
+- `jsql` [`<Object[]>`][object]
 
 ###### Callback arguments
 
-* `data` [`<Object[]>`][object], fetched data
+- `data` [`<Object[]>`][object], fetched data
 
 [errors]: https://github.com/metarhia/globalstorage/blob/master/lib/errors.js
-[`<Object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
+[`<object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
```
Comment (belochub) [inline doc/api/jstp.md]: I think you can remove this link and instead use `[object]` in all the related places.
```suggestion
```

File: doc/api/jstp.md [hunk 7]
```diff
@@ -25,11 +25,11 @@ Get a record by id.
 
 ###### Call arguments
 
-* `id` [`<string>`][], id of the record
+- `id` [`<string>`][], id of the record
 
 ###### Callback arguments
 
-* `record` [`<Object>`][]
+- `record` [`<Object>`][]
```
Comment (belochub) [inline doc/api/jstp.md]: As for the comment above:
```suggestion
- `record` [`<Object>`][object]
```

File: .prettierignore (hunk 5 above)
Comment (JuliaGerasymenko) [reply .prettierignore]: Because these files are generated automatically by using a script prepublish in package.json

Comment (belochub) [reply .prettierignore]: No, they are not automatically generated, also, all `prepublish` does is run `babel`:
https://github.com/metarhia/globalstorage/blob/81366e41f8d346b98cd7203b55a58348ce815d7d/package.json#L40-L41

File: doc/api/jstp.md [hunk 8]
```diff
@@ -54,8 +54,8 @@ Set a record, `Id` field must be present in the record for it to work.
 
 ###### Call arguments
 
-* `record` [`<Object>`][]
-  * `Id` [`<string>`][]
+- `record` [`<Object>`][]
```
Comment (belochub) [inline doc/api/jstp.md]: You forgot to specify a link here:
```suggestion
- `record` [`<Object>`][object]
```

File: doc/api/jstp.md [hunk 9]
```diff
@@ -69,12 +69,12 @@ Create a record in the specified category.
 
 ###### Call arguments
 
-* `category` [`<string>`][]
-* `record` [`<Object>`][]
+- `category` [`<string>`][]
+- `record` [`<Object>`][]
```
Comment (belochub) [inline doc/api/jstp.md]: Same here:
```suggestion
- `record` [`<Object>`][object]
```

File: doc/api/jstp.md [hunk 10]
```diff
@@ -85,13 +85,13 @@ category.
 
 ###### Call arguments
 
-* `category` [`<string>`][]
-* `query` [`<Object>`][]
-* `patch` [`<Object>`][]
+- `category` [`<string>`][]
+- `query` [`<Object>`][]
```
Comment (belochub) [inline doc/api/jstp.md]: Ditto.

File: package.json [hunk 11]
```diff
@@ -36,7 +36,8 @@
   ],
   "scripts": {
     "test": "npm run lint && metatests test/",
-    "lint": "eslint . --ignore-path .gitignore",
+    "lint": "eslint . --ignore-path .gitignore && prettier -c \"**/*.js\" \"**/*.json\" \"**/*.md\" \".*rc\"",
```
Comment (belochub) [inline package.json]: You should also add `**/*.yml` files here and below.

File: doc/api/jstp.md [hunk 12]
```diff
@@ -151,16 +151,14 @@ Select a record or multiple records in the category using provided jsql.
...
+- `jsql` [`<Object[]>`][object]
 
 ###### Callback arguments
 
-* `data` [`<Object[]>`][object], fetched data
+- `data` [`<Object[]>`][object], fetched data
 
 [errors]: https://github.com/metarhia/globalstorage/blob/master/lib/errors.js
-[`<Object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
-[object]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
```
Comment (belochub) [inline doc/api/jstp.md]: This should not be removed.

File: .prettierignore [hunk 13]
```diff
@@ -0,0 +1,3 @@
+data
+dist
+doc/api/jstp.md
```
Comment (belochub) [inline .prettierignore]: This should not be ignored.

File: doc/api/jstp.md [hunk 14]
```diff
@@ -151,12 +151,12 @@ Select a record or multiple records in the category using provided jsql.
...
+- `category` [`<string>`][]
+- `jsql` [`<Object[]>`][object]
 
 ###### Callback arguments
 
-* `data` [`<Object[]>`][object], fetched data
+- `data` [`<Object[]>`][object], fetched data
 
 [errors]: https://github.com/metarhia/globalstorage/blob/master/lib/errors.js
 [`<Object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
```
Comment (belochub) [inline doc/api/jstp.md]: This link is not used anymore, you can remove it.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/globalstorage/commit/74ac71f057a824218f2beeb4dbe4492ba806fb6f.

**Pull request closed without merging**